src/uucore/src/lib/mods/clap_localization.rs

```diff
@@ -120,167 +102,182 @@ pub fn handle_clap_error_with_exit_code(err: Error, util_name: &str, exit_code:
         }
-        ErrorKind::UnknownArgument => {
-            // Force localization initialization - ignore any previous failures
-            crate::locale::setup_localization(util_name).ok();
-
-            // Choose exit code based on utility name
-            let exit_code = match util_name {
-                // These utilities expect exit code 2 for invalid options
-                "ls" | "dir" | "vdir" | "sort" | "tty" | "printenv" => 2,
-                // Most utilities expect exit code 1
-                _ => 1,
-            };
+        _ => unreachable!("handle_display_errors called with non-display error"),
+    }
+}
 
+/// Handle UnknownArgument errors with localization and suggestions
+fn handle_unknown_argument_error(
+    err: Error,
+    util_name: &str,
+    maybe_colorize: impl Fn(&str, Color) -> String,
+) -> ! {
+    if let Some(invalid_arg) = err.get(ContextKind::InvalidArg) {
+        let arg_str = invalid_arg.to_string();
+        // Get localized error word with fallback
+        let error_word = translate!("common-error");
+
+        let colored_arg = maybe_colorize(&arg_str, Color::Yellow);
+        let colored_error_word = maybe_colorize(&error_word, Color::Red);
+
+        // Print main error message with fallback
+        let error_msg = translate!(
+            "clap-error-unexpected-argument",
+            "arg" => colored_arg.clone(),
+            "error_word" => colored_error_word.clone()
+        );
+        eprintln!("{error_msg}");
+        eprintln!();
+
+        // Show suggestion if available
+        if let Some(suggested_arg) = err.get(ContextKind::SuggestedArg) {
+            let tip_word = translate!("common-tip");
+            let colored_tip_word = maybe_colorize(&tip_word, Color::Green);
+            let colored_suggestion = maybe_colorize(&suggested_arg.to_string(), Color::Green);
+            let suggestion_msg = translate!(
+                "clap-error-similar-argument",
+                "tip_word" => colored_tip_word.clone(),
+                "suggestion" => colored_suggestion.clone()
+            );
+            eprintln!("{suggestion_msg}");
+            eprintln!();
+        } else {
             // For UnknownArgument, we need to preserve clap's built-in tips (like using -- for values)
             // while still allowing localization of the main error message
             let rendered_str = err.render().to_string();
-            let _lines: Vec<&str> = rendered_str.lines().collect();
-
-            if let Some(invalid_arg) = err.get(ContextKind::InvalidArg) {
-                let arg_str = invalid_arg.to_string();
 
-                // Get localized error word with fallback
-                let error_word = translate!("common-error");
+            // Look for other clap tips (like "-- --file-with-dash") that aren't suggestions
+            // These usually start with "  tip:" and contain useful information
+            for line in rendered_str.lines().collect::<Vec<_>>().iter() {
```

Comment:
I think you can simplify it because `lines()` already returns an iterator:
```suggestion
            for line in rendered_str.lines() {
```